app/scripts/background.js

```diff
@@ -475,6 +481,73 @@ function setupController(initState, initLangCode) {
     extension.browserAction.setBadgeBackgroundColor({ color: '#037DD6' });
   }
 
+  notificationManager.on(
+    NOTIFICATION_MANAGER_EVENTS.POPUP_CLOSED,
+    rejectUnapprovedNotifications,
+  );
+
+  function rejectUnapprovedNotifications() {
+    // Metamask can open a popup for a login, we need to check if
+    // we're on a confirmation page
+    if (controller.isUnlocked()) {
```

Comment:
I think we want to reject all confirmations even if the wallet is locked. This is critical for ensuring subsequent `request_account` requests trigger the popup to open.

Issue #10085 is what prompted me to write up the issue this PR is addressing, and that case is about users getting stuck after closing the popup window once while the wallet was locked. Once they do that, subsequent attempts by the site to request access to accounts were rejected because we're already processing that request. Interpreting the closing of the popup as a rejection ensures that the next attempt will open the popup again.

It might be unexpected for a user if this ends up rejecting something unexpectedly that they haven't even seen yet. Maybe there's something we can do to guard against that mistake. I don't really understand why a user would choose not to unlock the wallet in this case though, especially if they've just done something that triggered a confirmation that is important to them.